harness/determined/constants.py

```diff
@@ -62,3 +62,7 @@
 # Increasing this from a default of 30 is necessary when there is a
 # large number of machines.
 HOROVOD_GLOO_TIMEOUT_SECONDS = 240
+
+# The well-known locations of the executing container's STDOUT and STDERR.
+CONTAINER_SDTOUT = "/var/log/determined.log"
+CONTAINER_STDERR = "/var/log/determined_error.log"
```

Comment:
>  but it would reduce the number of lines of communication, which is worth it IMO.

can you explain this?

Reply:
Right now there's a bunch of info passed from the SubprocessLauncher to the SubprocessReceiver through the WorkerProcessContext, and this would introduce more information "passing" (ok, "hardcoding") which is spread throughout the entrypoint.sh of the master and the constants.py file.

But in general, there's no reason for these paths to be hardcoded, and if they were not hardcoded they would be very similar in terms of sender/receiver to the other information in the WorkerProcessContext.

Reply:
the part that was confusing is the comment about `number of lines of communication`, you are just looking to avoid two separate sources of information which makes sense.

Reply:
I (not-strongly) feel this information should be separate from the `WorkerProcessContext`, and really would rather `worker_process_wrapper.py` _not_ do anything except essentially `run_my_process >>/var/log/determined.log 2>>/var/log/determined_error.log` but in python. If we're going to load the `WorkerProcessContext`, instead of just constants, it starts to open the surface area of that little script more than I want.

Reply:
I'd even be ok without `from determined import constants` and just putting the constants in `worker_process_wrapper.py`.

Reply:
> it starts to open the surface area of that little script more than I want

I am less concerned with the complexity of that little script than I am with the complexity of how information is hardcoded in some places 
.

How about this: instead of passing it

Reply:
You know what?  I'd even be ok if you just calculated the value, like `/tmp/determined-stdout-<exp_id>-<trial_id>`.  Then you could pass it as a command line parameter in the horovodrun command, and it's not hardcoded anywhere or at any risk of causing a collision when we someday support multi-gpu local training mode.

Reply:
My 2cents is that leaving this is as a constant the way it is right now is a perfectly fine solution.

Reply:
> I am less concerned with the complexity of that little script than I am with the complexity of how information is hardcoded in too many places and passed around in a bunch of different ways during distributed training.

This a real constant, it will never be changed, so I'm really unsure why we'd want to pass a constant through a context. I don't see this as hardcoding information but just using a constant, which I think is fine. I think passing it through the WorkerProcessContext is _more_ complicated, as far as this comment is concerned, than just using a constant.

If we someday support multi-gpu local training, we can just as easily make this change then to pass a full blown context to the wrapper script or we can bypass the wrapper altogether in local mode (which I think make more sense than `/tmp/determined-stdout-...`; in local mode, users would probably expect output to the STDOU/STDERR of their terminal, not a log file in tmp).

> In fact, come to think of it, you don't need to symlink at all if you use the WorkerProcessContext.

I don't understand this, only in the entrypoint or harness process would you have context on the correct value for `$$`, so I think you need the simlink (unless you pass the proc id in the context, which seems way too complicated.

Reply:
Bottom line for me: if someone comes to this later, and tries to understand how logging is happening through the worker process container, I think as is would take 5 minutes, and using WorkerProcessContext would take an hour.

Additionally, it's so many more LOC that it becomes significantly more error prone.

Based on those two points, I want to leave it as it.

Reply:
> we can bypass the wrapper altogether in local mode (which I think make more sense ...

I can get behind this.

But you do need to change where this file is, since `/var/log` is not a writable location for non-root containers.  Might I suggest putting it somewhere in our little subtree that we overlay every training job (`/run/determined`)?  Let's do `/run/determined/train` which is for things that are specific to training tasks.